core-java-modules/core-java-jndi/src/test/java/com/baeldung/jndi/JndiUnitTest.java

```diff
@@ -0,0 +1,60 @@
+package com.baeldung.jndi;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.jndi.JndiTemplate;
+import org.springframework.mock.jndi.SimpleNamingContextBuilder;
+
+import javax.naming.*;
+import javax.sql.DataSource;
+
+import java.sql.SQLException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class JndiUnitTest {
+
+    private static InitialContext ctx;
+    private static DriverManagerDataSource ds;
+
+    @BeforeAll
+    public static void setUp() throws NamingException {
+        SimpleNamingContextBuilder builder = new SimpleNamingContextBuilder();
+        ds = new DriverManagerDataSource("jdbc:h2:mem:mydb");
+        builder.activate();
+
+        JndiTemplate jndiTemplate = new JndiTemplate();
+        ctx = (InitialContext) jndiTemplate.getContext();
+    }
+
+    @Test
+    public void testNameCreation() throws InvalidNameException {
```

Comment:
Can we rename the test methods to the `given_when_then` format of names? `givenACompositeName_whenAddingAnElement_thenNameIncludesIt` - or similar